plugin/pkg/auth/authenticator/password/keystone/keystone.go

```diff
@@ -40,23 +44,49 @@ func (keystoneAuthenticator *KeystoneAuthenticator) AuthenticatePassword(usernam
 		Password:         password,
 	}
 
-	_, err := openstack.AuthenticatedClient(opts)
+	_, err := AuthenticatedClient(opts, keystoneAuthenticator.caFile)
 	if err != nil {
-		glog.Info("Failed: Starting openstack authenticate client")
+		glog.Info("Failed: Starting openstack authenticate client:" + err.Error())
 		return nil, false, errors.New("Failed to authenticate")
 	}
 
 	return &user.DefaultInfo{Name: username}, true, nil
 }
 
+// AuthenticatedClient logs in to an OpenStack cloud found at the identity endpoint specified by options, acquires a
+// token, and returns a Client instance that's ready to operate.
+func AuthenticatedClient(options gophercloud.AuthOptions, caFile string) (*gophercloud.ProviderClient, error) {
+	client, err := openstack.NewClient(options.IdentityEndpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	config := &tls.Config{}
+	if caFile !="" {
+		roots, err := certutil.NewPool(caFile)
+		if err != nil {
+			return nil, err
+		}
+		config.RootCAs = roots
+	}
+	client.HTTPClient.Transport = &http.Transport{TLSClientConfig: config, }
```

Comment:
is this equivalent to the transport set up by default by `openstack.AuthenticatedClient` (with respect to proxy settings, dial settings, timeouts, etc)?